core/fullHistory/hashEpochStorer.go

```diff
@@ -0,0 +1,49 @@
+//go:generate protoc -I=proto -I=$GOPATH/src -I=$GOPATH/src/github.com/ElrondNetwork/protobuf/protobuf  --gogoslick_out=. hashEpoch.proto
+
+package fullHistory
+
+import (
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type hashEpochProcessor struct {
+	marshalizer marshal.Marshalizer
+	storer      storage.Storer
+}
+
+func newHashEpochStorer(storer storage.Storer, marshalizer marshal.Marshalizer) *hashEpochProcessor {
+	return &hashEpochProcessor{
```

Comment:
yes. these checks for nil are no needed because hashEpochStorer in created inside NewHistoryRepository method and before creation of hashEpochStorer are these checks.